src/Lean/Elab/Tactic/Basic.lean

```diff
@@ -51,6 +51,9 @@ instance : Monad TacticM :=
   let i := inferInstanceAs (Monad TacticM);
   { pure := i.pure, bind := i.bind }
 
+instance : MonadParentDecl TacticM where
+  getParentDeclName? := Term.getDeclName?
```

Comment:
Where do we actually use this instance?

Reply:
https://github.com/leanprover/lean4/blob/778edc5f1d85dd31b370afaceda4893ff95003ca/src/Lean/Elab/InfoTree/Main.lean#L373

Reply:
We do not need this instance, so I removed it.